src/components/Scanner.vue

```diff
@@ -34,7 +37,17 @@ export default {
     IonToolbar,
     StreamBarcodeReader,
   },
+  mounted() {
+    this.checkCameraPermission();
```

Comment:
As we are checking for permission on mounted of scanner component, that will run when the modal opens up, so even we don't have camera permission an empty modal will be displayed.

I think we can move this logic for permission check before opening the scanner modal.